src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectViewContainer.tsx

```diff
@@ -104,12 +105,16 @@ const ProjectViewContainer: React.FC<ProjectViewContainerProps> = ({ project, en
     </MenuItem>
   );
 
+  const assignToTeamText = project.teams.map((team) => team.teamId).includes(teamAsHeadId || '')
```

Comment:
Doing || '' here might cause issues; If teamAsHeadId is undefined you should have the button say Assign to My Team, but disable it (since there's no team to assign to), so do the undefined check, then the check to see if the teamid is one of the project's teams

Reply:
I dont think we show the button at all ifteamAsHeadId doesnt exist

Reply:
Then I'd say do this
```
  const AssignToMyTeamButton = () => {
    const assignToTeamText = project.teams.map((team) => team.teamId).includes(teamAsHeadId!)
      ? 'Unassign from My Team'
      : 'Assign to My Team';

    return (
      <MenuItem onClick={handleAssignToMyTeam}>
        <ListItemIcon>
          <GroupIcon fontSize="small" />
        </ListItemIcon>
        {assignToTeamText}
      </MenuItem>
    );
  };
```
and call it like `<AssignToMyTeamButton/>` below